contrib/cryptomb/private_key_providers/test/ops_test.cc

```diff
@@ -38,398 +33,343 @@ ssl_private_key_result_t rsaPrivateKeyDecryptForTest(CryptoMbPrivateKeyConnectio
-
-  EC_KEY* ec = PEM_read_bio_ECPrivateKey(bio.get(), nullptr, nullptr, nullptr);
-
-  RELEASE_ASSERT(ec != nullptr, "PEM_read_bio_ECPrivateKey failed.");
-  RELEASE_ASSERT(1 == EVP_PKEY_assign_EC_KEY(key.get(), ec), "EVP_PKEY_assign_EC_KEY failed.");
-  return key;
-}
+class TestCallbacks : public Envoy::Ssl::PrivateKeyConnectionCallbacks {
+public:
+  void onPrivateKeyMethodComplete() override{
 
-TEST(CryptoMbProviderTest, TestEcdsaSigning) {
-  Event::SimulatedTimeSystem time_system;
-  Stats::TestUtil::TestStore server_stats_store;
-  Api::ApiPtr api = Api::createApiForTest(server_stats_store, time_system);
-  Event::DispatcherPtr dispatcher(api->allocateDispatcher("test_thread"));
-  bssl::UniquePtr<EVP_PKEY> pkey = makeEcdsaKey();
-  std::shared_ptr<FakeIppCryptoImpl> fakeIpp = std::make_shared<FakeIppCryptoImpl>(true);
+  };
+};
 
-  CryptoMbQueue queue(std::chrono::milliseconds(200), KeyType::Ec, 256, fakeIpp, *dispatcher);
+class CryptoMbProviderTest : public testing::Test {
+protected:
+  CryptoMbProviderTest()
+      : api_(Api::createApiForTest(store_, time_system_)),
+        dispatcher_(api_->allocateDispatcher("test_thread")),
+        fakeIpp_(std::make_shared<FakeIppCryptoImpl>(true)),
+        stats_(generateCryptoMbStats("cryptomb", store_)) {}
+
+  bssl::UniquePtr<EVP_PKEY> makeRsaKey() {
+    std::string file = TestEnvironment::readFileToStringForTest(TestEnvironment::substitute(
+        "{{ test_rundir }}/contrib/cryptomb/private_key_providers/test/test_data/rsa-1024.pem"));
+    bssl::UniquePtr<BIO> bio(BIO_new_mem_buf(file.data(), file.size()));
+
+    bssl::UniquePtr<EVP_PKEY> key(EVP_PKEY_new());
+
+    RSA* rsa = PEM_read_bio_RSAPrivateKey(bio.get(), nullptr, nullptr, nullptr);
+    RELEASE_ASSERT(rsa != nullptr, "PEM_read_bio_RSAPrivateKey failed.");
+    RELEASE_ASSERT(1 == EVP_PKEY_assign_RSA(key.get(), rsa), "EVP_PKEY_assign_RSA failed.");
+    return key;
+  }
+
+  bssl::UniquePtr<EVP_PKEY> makeEcdsaKey() {
+    std::string file = TestEnvironment::readFileToStringForTest(TestEnvironment::substitute(
+        "{{ test_rundir "
+        "}}/contrib/cryptomb/private_key_providers/test/test_data/ecdsa-p256.pem"));
+    bssl::UniquePtr<BIO> bio(BIO_new_mem_buf(file.data(), file.size()));
+
+    bssl::UniquePtr<EVP_PKEY> key(EVP_PKEY_new());
+
+    EC_KEY* ec = PEM_read_bio_ECPrivateKey(bio.get(), nullptr, nullptr, nullptr);
+
+    RELEASE_ASSERT(ec != nullptr, "PEM_read_bio_ECPrivateKey failed.");
+    RELEASE_ASSERT(1 == EVP_PKEY_assign_EC_KEY(key.get(), ec), "EVP_PKEY_assign_EC_KEY failed.");
+    return key;
+  }
+
+  Stats::TestUtil::TestStore store_;
+  Api::ApiPtr api_;
+  Event::SimulatedTimeSystem time_system_;
+  Event::DispatcherPtr dispatcher_;
+  std::shared_ptr<FakeIppCryptoImpl> fakeIpp_;
+  CryptoMbStats stats_;
+
+  // Result of an operation.
+  ssl_private_key_result_t res_;
+
+  // A size for signing and decryption operation input chosen for tests.
+  static constexpr size_t IN_LEN = 32;
```

Comment:
Const member variables follow member-variable naming convention, not macro convention, so call these in_len_, in_, etc.

Reply:
Fixed.